src/ssl.c

```diff
@@ -13678,6 +13716,25 @@ int wolfSSL_GetSessionFromCache(WOLFSSL* ssl, WOLFSSL_SESSION* output)
     }
 #endif
 
+#if defined(WOLFSSL_TLS13) && defined(HAVE_SESSION_TICKET) &&                  \
+    defined(WOLFSSL_TICKET_NONCE_MALLOC)
+    if (output->ticketNonce.data != output->ticketNonce.dataStatic) {
+        XFREE(output->ticketNonce.data, output->heap,
+            DYNAMIC_TYPE_SESSION_TICK);
+        output->ticketNonce.data = output->ticketNonce.dataStatic;
+        output->ticketNonce.len = 0;
+    }
+    error = SessionTicketNoncePrealloc(&preallocNonce, &preallocNonceLen,
+        output->heap);
+    if (error != 0) {
+#ifdef WOLFSSL_SMALL_STACK
+        if (tmpTicket != NULL)
```

Comment:
Above `tmpBufSet` is set into `output->ticket` then free'd. Should the code above set `tmpBufSet = NULL` after transferring it to the `output`?

Reply:
Thanks for spotting this. I think we should add
```
        if (tmpBufSet) {
            output->ticket = output->staticTicket;
            output->ticketLenAlloc = 0;
        }
```
so we error out without leaving the dangling pointer in `output` (same as below).

Reply:
(added in the last version)